### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by 3h4x**
## what
* created opsgenie submodules
  - alert_policy        
  - api_integration     
  - escalation          
  - notification_policy 
  - team                
  - team_routing_rule
* every module is tested
* every module has example
* root module as an example module which use all submodules

## notes
- disabled superlinter workflow as terrascan error is related to not supporting HCL2 yet. Issue already exist and feature should be soonish released https://github.com/accurics/terrascan/issues/233

### CODE REVIEW

**Comment by osterman** [comment]
also add `versions.tf`

File: modules/alert_policy/alert_policy.tf [hunk 1]
```diff
@@ -0,0 +1,52 @@
+variable "alert_policy" {
```
Comment (danjbh) [inline modules/alert_policy/alert_policy.tf:1]: So just as a general pattern question -- why don't we put this in `variables.tf`?

And while we're at it, why do we we put this stuff in `alert_policy.tf` instead of `main.tf`? Just curious mostly..

Comment (3h4x) [reply modules/alert_policy/alert_policy.tf:1]: For consistency across our modules
https://github.com/cloudposse/terraform-datadog-monitor/tree/master/modules/cpu

I'm easy with both approaches and as `main.tf` doesn't have anything other than provider maybe we should change the convention and put everything in `main.tf`.

Comment (danjbh) [reply modules/alert_policy/alert_policy.tf:1]: I see.. I'm somewhat impartial but perhaps @osterman can chime in.

**Comment by danjbh** [comment]
I took a pass through all of the modules and spent a bit of time understanding each one -- it all looks pretty straight-forward to me!

File: modules/alert_policy/alert_policy.tf (hunk 1 above)
Comment (osterman) [reply modules/alert_policy/alert_policy.tf:1]: oh,`terraform-datadog-monitor` is probably the *worst* example to use =) it was never used in production, still on HCLv1, and has no tests. I don't believe any of our HCL2 modules with tests do this. 

By in large, we do try to follow the hashicorp recommended best practices for layouts - and if we don't let me know and we'll try to correct it.

File: modules/escalation/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+```
+
+## Inputs
+
+**Note:** `escalation` is a map for two reasons: 
+- to be able to put whole configuration in yaml file
+- variables defined with type set are not robust enough (can't set default values)
+
```
Comment (osterman) [inline modules/escalation/README.md:37]: Hrmmmmmm this is where the submodules cause scope creep. Our readme generation/validation stuff won't work. I know there's recommit stuff for that. Just it's another convention.

Comment (3h4x) [reply modules/escalation/README.md:37]: What do you mean exactly? Do you mean using map for whole escalation or are you referring to readme generation from yaml?

> I know there's recommit stuff for that.
Can you elaborate?

File: modules/alert_policy/alert_policy.tf (hunk 1 above)
Comment (3h4x) [reply modules/alert_policy/alert_policy.tf]: Reorganized repo.

**Comment by 3h4x** [comment]
My main concern is variable naming. I found redundant to name variable `alert_policy` with added prefix `opsgenie_` but maybe it would be preferred to do that. If such variable would be declared in external system I might find it confusing to see `api_integration` as not just `opsgenie` can provide it.
@osterman Any thoughts?

File: modules/escalation/README.md (hunk 2 above)
Comment (3h4x) [reply modules/escalation/README.md:37]: 👍
